pkg/otlp/attributes/source.go

```diff
@@ -29,6 +29,9 @@ const (
 	AttributeDatadogHostname = "datadog.host.name"
 	// AttributeK8sNodeName the datadog k8s node name attribute
 	AttributeK8sNodeName = "k8s.node.name"
+	// Attribute host is a literal host tag.
+	// We check for this to avoid double tagging.
+	AttributeHost = "host"
```

Comment:
IMO it is very confusing that we treat `host`, `host.name` and `datadog.host.name` in subtly different order. This has led to several customer issues.

(Not suggesting to fix it in this PR, but something that needs better documentation)

Reply:
That's fair, I think some subtlety is required here but maybe we can do better. For this PR, I am convinced that `host` needs to go first, since otherwise you get double tagging in some cases, so I will merge, but we should discuss how to improve the situation for the rest.